node/request.js

```diff
@@ -226,32 +228,50 @@ TChannelRequest.prototype.resend = function resend() {
         opts[key] = self.options[key];
     }
     opts.timeout = self.timeout - self.elapsed;
-    var outReq = peer.request(opts);
-    self.outReqs.push(outReq);
 
-    if (self.outReqs.length === 1) {
-        self.channel.outboundCallsSentStat.increment(1, {
-            'target-service': outReq.serviceName,
-            'service': outReq.headers.cn,
-            // TODO should always be buffer
-            'target-endpoint': String(self.arg1)
+    self.services.onRequest(self);
+    var conn = peer.connect();
+    if (!peer.isConnected() && conn.handler) {
+        var timer = self.timers.setTimeout(function connectTimeOut() {
```

Comment:
In order to keep this PR small, I will do a different PR to move the timeout check to TChannelConnectionBase.prototype.onTimeoutCheck.